### PR TITLE
Add `actions/labeler`

### PR DESCRIPTION
**Pull request by silverwind**
Implements https://github.com/GiteaBot/gitea-backporter/issues/93 using [`actions/labeler`](https://github.com/actions/labeler). Very basic configuration, can be extended later.

### CODE REVIEW

**Comment by silverwind** [comment]
It does not seem to run on `pull_request_target` which itself seems to be some [workaround](https://github.com/actions/labeler#permissions) related to permissions. Or maybe it will run on `pull_request_target` after merging only. I'll revert to it now because on `pull_request` fails to obtain the right permissions (likely needs a write token).

**Comment by silverwind** [comment]
I don't know why this [`pull_request_target`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target) does not run. Either it only works after this has merged or some permissions on our repo may prevent it from working.

**Comment by techknowlogick** [comment]
@silverwind just pushed a trivial change, hopefully it triggers now.

**Comment by silverwind** [comment]
For some reason the workflow doesn't even show up in the list. It did show up when I set `on: pull_request` but that ran into the mentioned permission error.

File: docs/README.md
Comment (lunny) [inline docs/README.md]: What's this?

Comment (silverwind) [reply docs/README.md]: Just a test to confirm that labeling works where it should add `kind/docs` because of this change. Will revert before merging of course.

**Comment by silverwind** [comment]
Maybe we actually need to merge this to work on subsequent PRs. From [docs](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request_target):

> This event runs in the context of the base of the pull request, rather than in the context of the merge commit, as the pull_request event does

File: docs/README.md
Comment (silverwind) [reply docs/README.md]: Reverted.

File: .github/labeler.yml
Comment (JakobDev) [inline .github/labeler.yml:6]: You should also include the templates

**Comment by silverwind** [comment]
So I'm pretty positive this can only work once merged. The action runs on the target branch with write access to github api to set labels. It obtains changed files from [API](https://github.com/actions/labeler/blob/a76cf44ddb1ab997b52718d59b0efc8adc55cf5c/src/labeler.ts#L143) and uses the config in the repo. But because the base repo (`main`) has no config or workflow yet, it does not work on the initial PR.

File: .github/labeler.yml
Comment (silverwind) [reply .github/labeler.yml:6]: Fixed in https://github.com/go-gitea/gitea/pull/26962/commits/570888975e16695c078a127d66fcdcaff51d33be. This config is far from complete, but I would first like to get it running in a basic state.

**Comment by silverwind** [comment]
Added some more, I think this is good enough now. We should see it working after merge.

File: .github/labeler.yml
Comment (KN4CK3R) [inline .github/labeler.yml]: If a PR imports a new module, this PR gets labeled as `dependencies`?

Comment (silverwind) [reply .github/labeler.yml]: Yes, at least that's my understanding of the label. It says "Pull requests that update a dependency file". We can of course remove the setting of this label entirely if it's not useful, but I do see the addition/modification of depdendencies a important property on a PR.

**Review by techknowlogick — APPROVED** [review]

File: .github/labeler.yml
Comment (silverwind) [reply .github/labeler.yml]: I have removed the `dependencies` now for two reasons:

- The label does not follow the `kind/` prefix
- I don't want to go overboard with label spam

**Review by KN4CK3R — APPROVED** [review]
Let's try it.

File: .github/labeler.yml
Comment (silverwind) [inline .github/labeler.yml:3]:
```suggestion
  - **/*.md
  - docs/**/*
```

File: .github/labeler.yml
Comment (silverwind) [inline .github/labeler.yml:17]:
```suggestion
  - docker/**
  - webpack.config.js
```

**Comment by silverwind** [comment]
Fixup: https://github.com/go-gitea/gitea/pull/26964

**Comment by silverwind** [comment]
It works:

image

https://github.com/go-gitea/gitea/pull/26970

**Pull request merged**
